### ISSUE TITLE
To install bosh on El Capitan, you need bundler build options

### ISSUE THREAD

**Issue by voelzmo** [cloudfoundry/bosh#1064]
doing a `bundle install` on a fresh El Capitan machine gives you

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /Users/cpi/.rvm/rubies/ruby-2.1.5/bin/ruby -r ./siteconf20151211-9136-g5kmpx.rb extconf.rb
checking for rb_trap_immediate in ruby.h,rubysig.h... no
checking for rb_thread_blocking_region()... yes
checking for inotify_init() in sys/inotify.h... no
checking for __NR_inotify_init in sys/syscall.h... no
checking for writev() in sys/uio.h... yes
checking for rb_wait_for_single_fd()... yes
checking for rb_enable_interrupt()... no
checking for rb_time_new()... yes
checking for sys/event.h... yes
checking for sys/queue.h... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling binder.cpp
In file included from binder.cpp:20:
./project.h:107:10: fatal error: 'openssl/ssl.h' file not found
#include <openssl/ssl.h>
         ^
1 error generated.
make: *** [binder.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/cpi/.rvm/gems/ruby-2.1.5/gems/eventmachine-1.0.3 for inspection.
Results logged to /Users/cpi/.rvm/gems/ruby-2.1.5/extensions/x86_64-darwin-14/2.1.0-static/eventmachine-1.0.3/gem_make.out
An error occurred while installing eventmachine (1.0.3), and Bundler cannot continue.
Make sure that `gem install eventmachine -v '1.0.3'` succeeds before bundling.
```

which means that the build can't find the openssl installed via `brew`

`brew info openssl` even hints you to the solution

```
This formula is keg-only, which means it was not symlinked into /usr/local.

Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries

Generally there are no consequences of this for you. If you build your
own software and it requires this formula, you'll need to add to your
build variables:

    LDFLAGS:  -L/usr/local/opt/openssl/lib
    CPPFLAGS: -I/usr/local/opt/openssl/include
```

However, we should probably put something in the development docs:

This is necessary to configure bundler to use a different path to find openssl:

```
bundle config --global build.eventmachine --with-cppflags=-I/usr/local/opt/openssl/include --with-ldflags=-L/usr/local/opt/openssl/lib
```

**Comment by cactis** [comment]
thanks!!

**Comment by voelzmo** [comment]
A more generic approach could also be to set `export PKG_CONFIG_PATH=$(brew --prefix openssl)/lib/pkgconfig` instead of setting bundler options. This should also install everything else that requires a link to openssl.

**Comment by ThadeuBrito** [comment]
Thanks!!

**Comment by mfine30** [comment]
@voelzmo do you know if this is still an issue? Can we close this?

**Comment by voelzmo** [comment]
Has been fixed with eventmachine 1.0.9.1. Closing this issue.
